### DISCUSSION TITLE
Flatnode file outside of container

### DISCUSSION THREAD

**Discussion by gsg-git** [mediagis/nominatim-docker#489] [Q&A]
Hello!

Is it possible to set the flatnode folder outside the container like the Nominatim -v nominatim-data flag?

like this maybe?
docker run...
-v nominatim-flatnode:/nominatim/flatnode \
-v /home/Nominatim_docker/flatnode:/nominatim/flatnode \

**Comment by philipkozeny** [comment]
Yes, like it is explained here: https://github.com/mediagis/nominatim-docker/blob/master/4.3/README.md#flatnode-files 

But please read how docker volumes work, from your example `-v nominatim-flatnode:/nominatim/flatnode` is a volume, `-v /home/Nominatim_docker/flatnode:/nominatim/flatnode \`is a bind mount. So I depends on what you want to achieve and use. You only need one of this options.

**Reply by gsg-git** [reply]
@philipkozeny 
Thank you, but i thought i need the -v nominatim-flatnode:/nominatim/flatnode at least for the flantode file to be generated and the other to mount the folder of the flatnode to the local system.
Or is there a diffrent flag in which i can get both in one?
Or does the install Script see the Flatnode flag and set the necessary flatnode file and settings?
I couldnt recognize anything like this as i looked the files in this git and searched for flatnode.

I just sumbled over the  flag to add local folder to the container as i interviewed the gpt bot how i get my data from container,
Because the container i have running has everything in itself.
I just found out too late that the recommended way of updating the Nominatim docker is to set a new container with the new version and move the data (ill try the other way anyway because im curious).
So now im evaluating how i can ease updates in the future with the database and flatnode placed local on the host, because i have imported the world.

**Reply by gsg-git** [reply]
In the example for persisted container data it seems the flatnode is not included, because not explicid mentioned like "with this example the flatnode is included" - at least for me

**Reply by philipkozeny** [reply]
Here's one of my docker-compose files in use in production, there you can see that I have two volumes mounted, one for the data, one for the flatnode file.

```docker
version: "3"

services:
    nominatim:
        container_name: nominatim
        image: mediagis/nominatim:4.3
        ports:
            - 10.0.0.30:8080:8080
        environment:
            PBF_URL: "https://download.geofabrik.de/europe-latest.osm.pbf"
            REPLICATION_URL: "https://download.geofabrik.de/europe-updates/"
            NOMINATIM_PASSWORD: "xxx"
            IMPORT_WIKIPEDIA: "true"
            UPDATE_MODE: "continuous"
            USER_AGENT: "xyz/nominatim"
        volumes:
            - /mnt/volume-nominatim/postgresql:/var/lib/postgresql/14/main
            - /mnt/volume-nominatim/data:/nominatim/flatnode
        shm_size: 1gb
 ```

**Reply by gsg-git** [reply]
Thank you!
have you created the flatnode file in the past and had to set it after the setup?
I cant see where in your setup the nominatim knows where the file is.
